File: XWA.UI/wwwroot/app/parkVisit/parkVisitService.js
angular.module('xwa.parkVisitService', [
  'ui.bootstrap',
  "xwa.common"
]).service('ParkVisitService', [
  '$http',
  'Common',
  function (
    $http,
    common) {

    this.getParkVisits = function () {
      var url = common.getServiceUrl('parkVisit/parkVisits', true);
      return $http.get(url)
        .then(function success(response) {
          return response;
        });
    };
  }]);

File: XWA.WebAPI/Features/ParkVisit/ParkVisitService.cs
using Microsoft.Extensions.Options;
using XWA.WebAPI.Features.Park;
using XWA.WebAPI.Features.Region;
using XWA.WebAPI.Options;

namespace XWA.WebAPI.Features.ParkVisit;

/// <summary>
/// The park visit service class.
/// </summary>
/// <param name="csvFileOptions">The CSV file path and name values.</param>
/// <param name="fipsOptions">The FIPS codes from the appsettings.json file.</param>
public class ParkVisitService(
    IOptions<CsvFileOptions> csvFileOptions,
    IOptions<FipsOptions> fipsOptions) : IParkVisitService
{
    /// <summary>
    /// The park visit service wrapper method.
    /// </summary>
    /// <returns>The park visit response model.</returns>
    public async Task<ParkVisitResponse> GetParkVisitAsync()
    {
        IList<RegionResponse> regionsPrototype = await GetRegions.Execute(csvFileOptions.Value.Regions);

        IList<ParkResponse> parksPrototype = await GetParks.Execute(csvFileOptions.Value.Parks, fipsOptions.Value.Oconus);

        return await GetParkVisits.Execute(regionsPrototype, parksPrototype);
    }
}
